src/sparseml/exporters/transforms/convert_quantizable_matmul.py

```diff
@@ -0,0 +1,159 @@
+    # (or follows the last optional node if optional nodes present)
+    for node_to_delete in [a_dequant, b_dequant, output_quant]:
+        delete_quant_node(model, node_to_delete)
+
+    # set dequantize's input to quant's input
+    # NOTE: this handles the presence of the optional transpose/reshape nodes
+    output_dequant.input[0] = output_quant.input[0]
+
+    # create qmatmul node and add it to graph
+    qmatmul_node = onnx.helper.make_node(
+        "QLinearMatMul",
+        qmatmul_inputs,
+        [match.node.output[0]],
+        "{}_quant".format(match.node.name),
+    )
+    model.graph.node.append(qmatmul_node)
+    return model
+
+
+class ConvertQuantizableMatmul(OnnxTransform):
+    """
+    A transform that attempts, if possible, to convert MatMul nodes into
+    their quantized representation.
+    This MatMul is the result of quantizing native torch.matmul using QATMatMul
+
+    | Starting with:
+    |          INPUT_0           INPUT_1
+    |            |               |
+    |     QuantizeLinear     QuantizeLinear
+    |            |               |
+    |     DequantizeLinear   DequantizeLinear
+    |                  |      |
+    |                   MatMul
+    |                     |
+    |                  Transpose (optional)
+    |                     |
+    |                  Reshape (optional)
+    |                     |
+    |               QuantizeLinear
+    |                     |
+    |              DequantizeLinear
+    |                     |
+    |                  OUTPUT
+    |
+    | We end up converting to:
+    |          INPUT_0           INPUT_1
+    |            |               |
+    |     QuantizeLinear     QuantizeLinear
+    |                  |      |
+    |                  |      |
+    |                   QLinearMatMul
+    |                     |
+    |                   Transpose (optional)
+    |                     |
+    |                    Reshape (optional)
+    |                     |
+    |              DequantizeLinear
+    |                     |
+    |                  OUTPUT
+    """
+
+    def transform(self, model: ModelProto) -> ModelProto:
+        graph = ONNXGraph(model)
+        count_converted_nodes = 0
+        for match in get_structural_matches(
+            graph,
+            parent_ops=[
+                ["QuantizeLinear", "DequantizeLinear"],
+                ["QuantizeLinear", "DequantizeLinear"],
+            ],
+            op_type="MatMul",
+            children_ops=[
+                [
+                    optional_node("Transpose"),
+                    optional_node("Reshape"),
+                    "QuantizeLinear",
+                    "DequantizeLinear",
+                ]
+            ],
+        ):
+            for quantize_linear_parent in [match.parents[0][0], match.parents[1][0]]:
+                if graph.get_init_by_name(quantize_linear_parent.input[0]):
+                    continue
```

Comment:
We should capture this in the docstring that it expects non initializers as parents